core/runtime/src/main/java/io/quarkiverse/langchain4j/guardrails/InputGuardrailResult.java

```diff
@@ -0,0 +1,112 @@
+package io.quarkiverse.langchain4j.guardrails;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class InputGuardrailResult {
+
+    private static final InputGuardrailResult SUCCESS = new InputGuardrailResult();
+
+    private final boolean success;
+    private List<Failure> failures;
```

Comment:
Since the validations are composable now, we accumulate all the failures here. This could be removed if we decide that we don't want composition in any case.

Reply:
I think we need both. 

It was designed to not do that but allows complex guards to used suppressed exceptions to give multiple validation issues (like a guard using bean validation, or a guard detecting offensive words (each of them in a suppressed). We need a way for a single guard to pass multiple potential failures.